virtual_ecosystem/core/base_model.py

```diff
@@ -254,10 +353,28 @@ def setup(self) -> None:
     def spinup(self) -> None:
         """Function to spin up the model."""
 
-    @abstractmethod
     def update(self, time_index: int, **kwargs: Any) -> None:
         """Function to update the model.
 
+        If the model is static, the inner update method, self._update will only run
+        once, at most.
+
+        Args:
+            time_index: The index representing the current time step in the data object.
+            **kwargs: Further arguments to the update method.
+        """
+        if self._static:
+            if not self._run_update:
+                return
+            else:
+                self._run_update = False
```

Comment:
This feels like a bit of a hack, but I'm not sure of a cleaner way to do things :shrug:.

Is there a way we could combine the initialisation and initial update steps for static models? Do we do anything with models while they're in an initialised-but-not-yet-updated state? Just a thought.

Reply:
We initialise the other models, which might provide variables that the update methods require.

Why is hacky? I thought I was being clever 😆 . Jokes aside, I don't see another way of running the inner `_update` just once (or none) for the static models. Amy suggestion is most welcome.

Reply:
I think the only issue I have here is the name `_run_update`, which could be a little more explicit? `_run_initial_static_update` or something?